contrib/src/main/java/com/datatorrent/contrib/cassandra/AbstractUpsertOutputOperator.java

```diff
@@ -0,0 +1,1401 @@
+    Metrics.Errors errors = session.getCluster().getMetrics().getErrorMetrics();
+    totalIgnoresInThisWindow = errors.getIgnores().getCount() - totalIgnoresSinceStart;
+    totalIgnoresSinceStart = errors.getIgnores().getCount();
+    totalWriteTimeoutsInThisWindow = errors.getWriteTimeouts().getCount() - totalWriteTimeoutsSinceStart;
+    totalWriteTimeoutsSinceStart = errors.getWriteTimeouts().getCount();
+    totalWriteRetriesInThisWindow =  errors.getRetriesOnWriteTimeout().getCount() - totalWriteRetriesSinceStart;
+    totalWriteRetriesSinceStart = errors.getRetriesOnWriteTimeout().getCount();
+    try {
+      // we do not need any particular state and hence reusing the window id itself
+      windowDataManager.save(currentWindowId,currentWindowId);
+    } catch (IOException e) {
+      LOG.error("Error while persisting the current window state " + currentWindowId + " because " + e.getMessage());
+      throw new RuntimeException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void committed(long windowId)
+  {
+    if (null != windowDataManager) {
+      try {
+        windowDataManager.committed(windowId);
+      } catch (IOException e) {
+        LOG.error("Error while committing the window id " + windowId + " because " + e.getMessage());
+        throw new RuntimeException(e.getMessage());
+      }
+    }
+  }
+
+  @Override
+  public void beforeCheckpoint(long windowId)
+  {
+    // nothing to be done here. Prevent concrete implementations to be forced to implement this
+  }
+
+  @Override
+  public void checkpointed(long windowId)
+  {
+    // Nothing to be done here. Concrete operators can override if needed.
+  }
+
+  /**
+   * Initializes cassandra cluster connections as specified by the Connection State manager.
+   * Aspects that are initialized here include Identifying primary key column names, non-primary key columns,
+   * collection type columns, counter columns
+   * It also queries the Keyspace and Table metadata for extracting the above information.
+   * It finally prepares all possible prepared statements that can be executed in the lifetime of the operator
+   * for various permutations like APPEND/REMOVE COLLECTIONS , LIST APPEND/PREPEND , set nulls, set TTLs etc
+   * @param context The apex framework context
+   */
+  @Override
+  public void activate(Context.OperatorContext context)
+  {
+    ConnectionStateManager.ConnectionBuilder connectionBuilder = withConnectionBuilder();
+    checkNotNull(connectionBuilder, " Connection Builder cannot be null.");
+    connectionStateManager = connectionBuilder.initialize();
+    cluster = connectionStateManager.getCluster();
+    session = connectionStateManager.getSession();
+    keyspaceName = connectionStateManager.getKeyspaceName();
+    tableName = connectionStateManager.getTableName();
+    checkNotNull(session, "Cassandra session cannot be null");
+    tuplePayloadClass = getPayloadPojoClass();
+    columnDefinitions = new HashMap<>();
```

Comment:
will remove dataTypeGroupedColumns and pojoFieldNamesToCassandraColumns.  

columnDefinitions represents the DataType mapping to a name. This is used in every tuple execution wherein we extract the correct getter for every pojo field that is processed. It is also used during the phase of metadata generation.

pojoFieldNamesToCassandraColumns is used to